src/Libraries/CoreNodeModels/Input/ColorPalette.cs

```diff
@@ -25,18 +29,46 @@ namespace CoreNodeModels.Input
     [OutPortNames("Color")]
     [OutPortTypes("Color")]
     [OutPortDescriptions("Selected Color.")]
-    public class ColorPalette : NodeModel
+    public class ColorPalette : NodeModel, INotifyPropertyChanged
     {
-        private DSColor dscolor = DSColor.ByARGB(255,0,0,0);
-        public DSColor dsColor
+        private DSColor dscolor = DSColor.ByARGB(255, 0, 0, 0);
+        private bool Isundo = false;
+        public DSColor DsColor
         {
             get { return dscolor; }
             set
-            {                              
+            {
+                if (dscolor.Equals(value))
+                {
+                    //Checks if undorecorder is being used
+                    Isundo = true;
+                }
                 dscolor = value;
+                if (!Isundo)
+                {
+                    //If undo recorder is not being used updates the undorecorder stack with new value
+                    Isundo = false;
+                    Update = dscolor;
+                }
                 OnNodeModified();
+                OnPropertyChanged();
+            }
+        }
+
+        public DSColor Update
+        {
+            set
+            {
+                OnPropertyChanged();
             }
         }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
```

Comment:
while this is nice, lets try to match all other onPropertyChanged handlers in the repo and just make this the property name string, drop the attribute, and just use the string and raise this directly with the string as an explicit parameter.